catalog/catalog.go

```diff
@@ -144,6 +150,7 @@ type options struct {
 	sigv4Service      string
 	prefix            string
 	authUri           *url.URL
+	scope             *string
```

Comment:
could this be a string and not a pointer? just use the zero-value (empty string) to indicate that we should use the default?

Reply:
Yea, I can change it. Intent was to differentiate between empty and not set

Reply:
Is there any particular benefit to differentiating that? I don't think that an empty scope is allowed by OAuth, right?

Reply:
> Is there any particular benefit to differentiating that? 

Likely not. I've pushed a commit to fix